### ISSUE TITLE
Help Wanted: Code block formatting is inconsistent

### ISSUE THREAD

**Issue by cwarnermm** [mattermost/docs#7066]
Code blocks in the documentation are formatted inconsistently. Some of them have a background, some of them not.

Example:
[Install Mattermost on Kubernetes - Mattermost documentation](https://docs.mattermost.com/install/install-kubernetes.html#with-helm)

Than we have these code blocks here: 
[Install Mattermost via Docker - Mattermost documentation](https://docs.mattermost.com/install/install-docker.html)
Which are shown without a background.

Sometimes they should have a background for the code blocks, like in [Install Mattermost on Kubernetes - Mattermost documentation](https://docs.mattermost.com/install/install-kubernetes.html#deploy-mattermost) point 4, but the one in the helm chart seems to be off.

![image (19)](https://github.com/mattermost/docs/assets/74422101/b71ed183-2106-41f9-990d-8fccb25148f8)
![image (18)](https://github.com/mattermost/docs/assets/74422101/2a693500-ace2-4d66-864a-3d63f878bf03)

**Comment by cwarnermm** [comment]
On this page [Important Upgrade Notes - Mattermost documentation](https://docs.mattermost.com/upgrade/important-upgrade-notes.html), the SQL syntaxes are formatted quite inconsistently. Some lines of the code are written as inline code whereas others are enclosed within code blocks and yet, some others are enclosed with both of them. Some occurrences can be at the upgrade note sections of v7.9, v7.8, etc.

![Screenshot 2024-04-13 004748 (1)](https://github.com/mattermost/docs/assets/74422101/0553ec57-c29b-437a-b7a8-b114d4846900)

**Comment by chessmadridista** [comment]
Hi, may I take up this issue? It seems like a straightforward change.

**Comment by cwarnermm** [comment]
Hi, @chessmadridista! The goal of this issue is to standardize on code blocks across all pages of the docs set. How would you approach this issue to standardize the code blocks?

**Comment by chessmadridista** [comment]
We need to standardize the code blocks across the entire documentation. All code blocks will have a background to maintain uniformity and to improve readability.

I will take the following steps:
1. Create an Excel sheet documenting which pages contain code snippets.
1.1. For each page, we will have the following columns:
- `(col_1)` Link: the link of the page.
- `(col_2)` Does it contain code snippets?: `Y/N` values.
- `(col_3)` Is everything properly formatted?: `Y/N` values for checking whether every code block in it is properly formatted (indentations, for example).
- `(col_4)` Do the code blocks have a background?: `Y/N` values for checking whether every code block in it has a background.
- `(col_5)` Are all the corrections done?: The cells will be filled with green background color to track the status of each page.
2. For every row marked `Y` for `col_2` and `N` for `col_3` or `col_4, I will make the necessary changes.
3. Once the changes are done, I will skim through the links to see whether every code block has a background and whether everything is properly formatted (indentations, for example).
4. After reviewing, I will create a PR.

As to how to add the background and do the proper formatting, I will look into the code blocks which have already have backgrounds and proper formatting and will replicate the functionality in places wherever it is needed.

**Comment by cwarnermm** [comment]
@chessmadridista - This is a great approach. As you dig further into the source code driving the code blocks, you'll find that code blocks can differ by type of code, and that not all types of code blocks have the same underlying CSS driving their look and feel. You'll want to take this into consideration in your tracking spreadsheet too.

**Comment by chessmadridista** [comment]
Hi @cwarnermm, [here](https://docs.google.com/spreadsheets/d/1H7lxDfczt-YitJUch_xEt9zqWhCoSR4AKXhrHiIsXZg/edit?usp=sharing) is the sheet. 

Please let me know if I need to add any other columns that I might have missed.

**Comment by chessmadridista** [comment]
Hi @cwarnermm, I have added the following columns as discussed.
- Col F: `Current code format syntax for the code blocks`.
- Col G: `Updated code format syntax for the code blocks`.
- Col J: `PR #`.

**Comment by cwarnermm** [comment]
Thanks, @chessmadridista! Similar to my feedback on https://github.com/mattermost/docs/issues/6658#issuecomment-2178795997, let's track source file name instead of page title. And perhaps we don't need column I, given that the spreadsheet is tracking granular detail already in previous columns.

If a given docs page doesn't include any code snippets, perhaps we don't include it in this spreadsheet (which eliminates the need for column B).

**Comment by chessmadridista** [comment]
> Thanks, @chessmadridista! Similar to my feedback on [#6658 (comment)](https://github.com/mattermost/docs/issues/6658#issuecomment-2178795997), let's track source file name instead of page title. And perhaps we don't need column I, given that the spreadsheet is tracking granular detail already in previous columns.
> 
> If a given docs page doesn't include any code snippets, perhaps we don't include it in this spreadsheet (which eliminates the need for column B).

Noted @cwarnermm, apologies for the late reply as I was away on vacation.

I have made the changes as suggested. I will start working on this now.

**Comment by chessmadridista** [comment]
@cwarnermm, I have a design question. Won't having a background for code snippets obfuscate the reading material and hamper the UX? I mean that the code snippets are already highlighted in themselves. So is there any particular pain point that we would be solving by adding the background? Or would removing the backgrounds be better?

**Comment by cwarnermm** [comment]
That's a good question, @chessmadridista. @matthewbirtch and UX Team, I'd love to hear your thoughts on this.

**Comment by matthewbirtch** [comment]
I see we have doubled-up background in a lot of places. I'm not seeing the need for it. We should avoid having two backgrounds wherever possible. It wastes space and doesn't seem to really help anything. Just keep the background on the code itself and remove the outer container's background.

So instead of this:
<img width="1147" alt="image" src="https://github.com/user-attachments/assets/40f477f7-37aa-423c-a217-aa27a50f6cab">

Do this:
<img width="835" alt="image" src="https://github.com/user-attachments/assets/33ab5e20-89fc-44c7-a61c-a34e7b4c7679">

**Comment by chessmadridista** [comment]
Hi @matthewbirtch, got it. Thank you for the clarification! 

@cwarnermm, do you think I should proceed with the proposed changes?

**Comment by cwarnermm** [comment]
I completely agree with @matthewbirtch's input and feedback. Thanks, Matt! @chessmadridista - please implement Matt's changes as described.

**Comment by CBID2** [comment]
I can pick up where the previous contributor has left off @cwarnermm

**Comment by cwarnermm** [comment]
Thanks, @CBID2! Looking forward to working with you on this!
